src/main.rs

```diff
@@ -0,0 +1,269 @@
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+use fxhash::FxHashMap;
+use regex::Regex;
+use std::collections::hash_map::Entry;
+use std::env;
+use std::fs;
+use std::io::{self, BufRead, Write};
+use symbolic_common::Name;
+use symbolic_debuginfo::Object;
+use symbolic_demangle::{Demangle, DemangleFormat, DemangleOptions};
+
+#[cfg(test)]
+mod tests;
+
+/// An interned string type. Many file paths are repeated, so having this type
+/// reduces peak memory usage significantly.
+#[derive(Clone, Copy)]
+struct InternedString(usize);
+
+/// A simple string interner. Each string is sub-optimally stored twice, but
+/// there is so much repetition relative to the number of distinct strings that
+/// this barely matters, and doing better would require use of `unsafe`.
+#[derive(Default)]
+struct Interner {
+    map: FxHashMap<String, usize>,
+    strings: Vec<String>,
+}
+
+impl Interner {
+    fn intern(&mut self, string: String) -> InternedString {
+        let index = if let Some(&index) = self.map.get(&string) {
+            index
+        } else {
+            let index = self.strings.len();
+            self.map.insert(string.clone(), index);
+            self.strings.push(string);
+            index
+        };
+        InternedString(index)
+    }
+
+    pub fn get(&self, interned_string: InternedString) -> &str {
+        &self.strings[interned_string.0]
+    }
+}
```

Comment:
A quick search on crates.io turned up the [string-interner](https://crates.io/crates/string-interner) crate. Maybe it's worth considering it for a future version?

Reply:
I looked at that previously but it seemed like overkill for this very simple usage. If things get more complicated later it might be worth switching.

Reply:
> 
> 
> I looked at that previously but it seemed like overkill for this very simple usage. If things get more complicated later it might be worth switching.

+1